src/server/api/tcp-interface.ts

```diff
@@ -146,6 +161,15 @@ export class BotTunnel {
         console.log({ msg });
         this.socket.write(msg);
     }
+
+    async waitForActionResponse(): Promise<void> {
+        return new Promise((res, rej) => {
+            this.emitter.once("actionComplete", (args) => {
+                if (args.success) res();
+                else rej(args.reason);
```

Comment:
I think we should always use brackets for readability. I'd imagine there's also an eslint rule that could do it automatically?